clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/config/MemoryEventRepositoryConfig.kt

```diff
@@ -0,0 +1,100 @@
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.validation.annotation.Validated
+import java.time.Duration
+import javax.validation.Constraint
+import javax.validation.ConstraintValidator
+import javax.validation.ConstraintValidatorContext
+import javax.validation.constraints.Min
+import kotlin.reflect.KClass
+
+@Configuration
+@ConditionalOnProperty("spinnaker.clouddriver.eventing.memory-repository.enabled", matchIfMissing = true)
+@EnableConfigurationProperties(MemoryEventRepositoryConfigProperties::class)
+open class MemoryEventRepositoryConfig {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  init {
+    log.info("Configuring EventRepository: MemoryEventRepository")
+  }
+
+  @Bean
+  open fun eventRepository(
+    properties: MemoryEventRepositoryConfigProperties,
+    eventPublisher: EventPublisher,
+    registry: Registry
+  ): EventRepository =
+    MemoryEventRepository(properties, eventPublisher, registry)
+}
+
+@MemoryEventRepositoryConfigProperties.SpinValidated
+@ConfigurationProperties("spinnaker.clouddriver.eventing.memory-repository")
+open class MemoryEventRepositoryConfigProperties {
+  /**
+   * The max age of an [Aggregate]. One of this and [maxAggregatesCount] must be set.
+   */
+  @Min(
+    message = "Event repository aggregate age cannot be less than 24 hours.",
+    value = 60 * 60 * 24 * 1000
+  )
+  var maxAggregateAgeMs: Long? = Duration.ofHours(24).toMillis()
+
+  /**
+   * The max number of [Aggregate] objects. One of this and [maxAggregateAgeMs] must be set.
+   */
+  var maxAggregatesCount: Int? = null
+
+  @Validated
+  @Constraint(validatedBy = [Validator::class])
+  @Target(AnnotationTarget.CLASS)
+  annotation class SpinValidated(
+    val message: String = "Invalid event repository configuration",
+    val groups: Array<KClass<out Any>> = [],
+    val payload: Array<KClass<out Any>> = []
+  )
+
+  class Validator : ConstraintValidator<SpinValidated, MemoryEventRepositoryConfigProperties> {
+    override fun isValid(
+      value: MemoryEventRepositoryConfigProperties,
+      context: ConstraintValidatorContext
+    ): Boolean {
+      if (value.maxAggregateAgeMs != null && value.maxAggregatesCount != null) {
+        context.buildConstraintViolationWithTemplate("Only one of 'maxAggregateAgeMs' and 'maxAggregatesCount' can be defined")
+          .addConstraintViolation()
+        return false
+      }
+      if (value.maxAggregateAgeMs == null && value.maxAggregatesCount == null) {
+        context.buildConstraintViolationWithTemplate("One of 'maxAggregateAgeMs' and 'maxAggregatesCount' must be set")
+          .addConstraintViolation()
+        return false
+      }
+      return true
+    }
+  }
+}
```

Comment:
Here's me, making an attempt at some better patterns for handling configuration within Spinnaker.